### PR TITLE
Fix `SymbolBlock` serde issue

### LINKED ISSUE #2180: A demo symbol block cannot run inference after being serialized and deserialized
## Description
Hello, 
I tried to serialize my predictor as symbo block format,
but when I deserialize them and use for predition, I got some error show below

## To Reproduce

```python
# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License").
# You may not use this file except in compliance with the License.
# A copy of the License is located at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# or in the "license" file accompanying this file. This file is distributed
# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
# express or implied. See the License for the specific language governing
# permissions and limitations under the License.

"""
This example shows how to fit a model and evaluate its predictions.
"""
import pprint
from functools import partial

import pandas as pd

from gluonts.dataset.repository.datasets import get_dataset
# from gluonts.mx.distribution.piecewise_linear import PiecewiseLinearOutput
from gluonts.evaluation import make_evaluation_predictions, Evaluator
from gluonts.model.deepar import DeepAREstimator
# from gluonts.model.transformer import TransformerEstimator
# from gluonts.mx import MQCNNEstimator
from gluonts.mx.trainer import Trainer
import os
import mxnet as mx
import numpy as np
from gluonts.mx.model import predictor as mxPredictor
from gluonts.model.predictor import Predictor as gPredictor

from pathlib import Path

from gluonts.mx.util import import_symb_block

datasets = [
    "m4_hourly",
    # "m4_daily",
    # "m4_weekly",
    # "m4_monthly",
    # "m4_quarterly",
    # "m4_yearly",
]

epochs = 10
num_batches_per_epoch = 5
estimators = [
    # MQCNNEstimator,
    DeepAREstimator,
    # partial(
    #     DeepAREstimator,
    #     distr_output=PiecewiseLinearOutput(8),
    # ),
]


def evaluate(dataset_name, estimator):
    dataset = get_dataset(dataset_name)
    print("got dataset")
    estimator = estimator(
        prediction_length=dataset.metadata.prediction_length,
        freq=dataset.metadata.freq,
        # use_feat_static_cat=True,
        # cardinality=[
        #     feat_static_cat.cardinality
        #     for feat_static_cat in dataset.metadata.feat_static_cat
        # ],
        trainer=Trainer(
            epochs=epochs,
            num_batches_per_epoch=num_batches_per_epoch,
        ),
    )

    predictor = estimator.train(dataset.train)

    symbol_predictor:mxPredictor.SymbolBlockPredictor = predictor.as_symbol_block_predictor(dataset=dataset.train)
    symbol_predictor.serialize(Path("/home/carkham/gluon-ts/gluon-ts-dev/examples/output"))
    print("exported model")

    new_symbol_predictor = mxPredictor.SymbolBlockPredictor.deserialize(Path("/home/carkham/gluon-ts/gluon-ts-dev/examples/output"))
    print("load predictor success")
    forecast_it, ts_it = make_evaluation_predictions(
        dataset.test, predictor=new_symbol_predictor, num_samples=100
    )

    agg_metrics, item_metrics = Evaluator()(
        ts_it, forecast_it, num_series=len(dataset.test)
    )
    return


if __name__ == "__main__":

    results = []
    for dataset_name in datasets:
        for estimator in estimators:
            # catch exceptions that are happening during training to avoid failing the whole evaluation
            try:
                results.append(evaluate(dataset_name, estimator))
            except Exception as e:
                print(str(e))

```

## Error message or code output

```
Parameter 'deeparpredictionnetwork0_featureembedder0_cat_0_embedding_weight' has not been initialized. Note that you should initialize parameters and create Trainer with Block.collect_params() instead of Block.params because the later does not include Parameters of nested child Blocks
```

## Environment
- Operating system:Ubuntu 20.04
- Python version:3.9.12
- GluonTS version:0.10.2
- MXNet version:1.9.0

(Add as much information about your environment as possible, e.g. dependencies versions.)

### PR DESCRIPTION
**Pull request by lostella**
*Description of changes:* This fixes #2180, but I'm not sure how and why, so I'm opening as a draft

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Comment by lostella** [comment]
Alternatively, the fix could be applied to the `as_symbol_block_predictor` method here: https://github.com/awslabs/gluon-ts/blob/08b6e2263ea5841c469a6624532133c81c45f1b1/src/gluonts/mx/model/predictor.py#L328-L331

**Review by jaheba — APPROVED** [review]
